base/include/AudioToTextXForm.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "Module.h"
+
+// size of audio to process should be a parameter. 
+// Cache variable to collect frames for processing
+
+class AudioToTextXFormProps : public ModuleProps
+{
+public:
+	enum DecoderSamplingStrategy {
+		GREEDY,      //WHISPER_SAMPLING_GREEDY
+		BEAM_SEARCH //WHISPER_SAMPLING_BEAM_SEARCH
+	};
+	AudioToTextXFormProps(
```

Comment:
This is also how it is in https://github.com/Apra-Labs/ApraPipes/blob/main/base/include/BrightnessContrastControlXform.h. I actually used that module as a reference.

Reply:
well it is a bad idea to put any method bodies in the header files because they get included at multiple places. all implementatons should always be in the cpp. when the function implementation changes the clients should not have to be recompiled. 
We may have missed this in previous reviews but we should improve it.

Reply:
Devil's advocate take: The module props are not frequently changed.

Reply:
@mraduldubey the basic principal of keeping header files clean should be followed either way